docs/features/software-templates/experimental.md

```diff
@@ -13,7 +13,66 @@ Please leave feedback on these features in the [Backstage Discord](https://disco
 
 ## Retries and Recovery
 
-### TODO
+Do you have long-running tasks, and can they recover and proceed after Backstage redeploy?
+
+Then you definitely will benefit from enabling the experimental scaffolder task recovery feature.
+
+Whenever you do redeploy, on startup there will be a check of all tasks in "processing" state that you identified in your template as they are capable of starting over.
+
+More details about the motivation and the goal of this feature you can read in [BEP "Scaffolder Retries and Idempotency"](https://github.com/backstage/backstage/tree/master/beps/0004-scaffolder-task-idempotency)
+
+This is an example of how you can do it:
+
+```yaml
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: recoverable-template
+spec:
+  EXPERIMENTAL_recovery:
+    EXPERIMENTAL_strategy: startOver
+```
+
+To enable the recovery feature, add this snippet into your app-config.yaml file:
+
+```yaml
+scaffolder:
+  EXPERIMENTAL_recoverTasks: true
+```
+
+By default, the tasks that are in a processing state and have a heartbeat of more than 30 seconds will be automatically recovered.
+This implies that the task's status will shift to "Open," initiating its execution from the beginning. So you have to be sure that your tasks can run multiple times.
+You can look at how to incorporate [checkpoints](https://backstage.io/docs/features/software-templates/writing-custom-actions#using-checkpoints-in-custom-actions-experimental) into your custom actions to achieve that.
+
+In case 30 seconds of no heartbeat time is not appropriate for your case to restart the task, you can customize it for your needs with the configuration:
+
+```yaml
+scaffolder:
+  EXPERIMENTAL_recoverTasksTimeout: { minutes: 1 }
+```
+
+If your task works with the filesystem, you might require serialization of your workspace.
+
+You can enable this feature with:
+
+```yaml
+scaffolder:
+  EXPERIMENTAL_workspaceSerialization: true
+```
+
+By default, the serialized workspace will be stored in your database.
+
+If you work with large files, it might not be the best option for you.
+
+At this moment we support the integration with the GCP bucket; to switch the serialization to this provider, you can with:
+
+```yaml
+scaffolder:
+  EXPERIMENTAL_workspaceSerializationProvider: gcpBucket
+  EXPERIMENTAL_workspaceSerializationGcpBucketName: name-of-your-bucket
```

Comment:
Ooh these two were probably unfortunately named. The product name is `GCS` or Cloud Storage; GCP is the entire platform. Like S3 vs AWS. I'd recommend changing these two in code and perhaps providing transparent backwards compatibility with these config key names too but mark them deprecated.

Even better perhaps:

```yaml
scaffolder:
  EXPERIMENTAL_workspaceSerializationProvider:
    type: gcs
    bucketName: name-of-your-bucket
```

then you can trivially add more providers with a bit nicer structure (it's also a pattern we use in several other places) and TypScript being more like `{ type: a, ...} | { type: b, ...}`

Reply:
Make sense. 

Yeah, I was thinking of platform naming and then the feature. 

Ok, I'll change to this structure!

Reply:
@freben Just realised that if we will change to 

```
scaffolder:
  EXPERIMENTAL_workspaceSerializationProvider:
    type: gcs
    bucketName: name-of-your-bucket
```

we can do it only without deprecation, as this variable is already used as a string.

This feature is experimental, so what is better to do, to use alternative name to `EXPERIMENTAL_workspaceSerializationProvider` or to introduce a breaking change?

Reply:
I think that we can rename this when we move off experimental, let's not worry about this too much right now.